Podverse/SearchPodcast.swift

```diff
@@ -0,0 +1,230 @@
+                let searchEpisode = SearchEpisode.convertJSONToSearchEpisode(episode)
+                searchPodcast.searchEpisodes.append(searchEpisode)
+            }
+        }
+        
+        return searchPodcast
+        
+    }
+    
+    static func retrievePodcastFromServer(id: String?, completion: @escaping (_ podcast:SearchPodcast?) -> Void) {
+        
+        if let id = id {
+            
+            if let url = URL(string: BASE_URL + "api/podcasts?id=" + id) {
+                var request = URLRequest(url: url, cachePolicy: NSURLRequest.CachePolicy.reloadIgnoringLocalAndRemoteCacheData, timeoutInterval: 60)
+                request.httpMethod = "GET"
+                
+                let task = URLSession.shared.dataTask(with: request) { data, response, error in
+                    
+                    guard error == nil else {
+                        DispatchQueue.main.async {
+                            completion(nil)
+                        }
+                        return
+                    }
+                    
+                    if let data = data {
+                        do {
+                            var searchPodcast = SearchPodcast()
+                            
+                            if let podcastJSON = try JSONSerialization.jsonObject(with: data, options: []) as? [String:Any] {
+                                searchPodcast = convertJSONToSearchPodcast(podcastJSON)
+                            }
+                            
+                            DispatchQueue.main.async {
+                                completion(searchPodcast)
+                            }
+                            
+                        } catch {
+                            print("Error: " + error.localizedDescription)
+                        }
+                    }
+                    
+                }
+                
+                task.resume()
+            }
+            
+        } else {
+            completion(nil)
+        }
+        
+    }
+    
+    static func authorityFeedUrlForPodcast(id: String, completion: @escaping (_ podverseId: String?) -> Void) {
+        
+        if let url = URL(string: BASE_URL + "api/podcasts/authorityFeedUrl?id=" + id) {
+            
+            var request = URLRequest(url: url, cachePolicy: NSURLRequest.CachePolicy.reloadIgnoringLocalAndRemoteCacheData, timeoutInterval: 60)
+            request.httpMethod = "GET"
+            
+            let task = URLSession.shared.dataTask(with: request) { data, response, error in
+                
+                guard error == nil else {
+                    print(error)
+                    DispatchQueue.main.async {
+                        completion(nil)
+                    }
+                    return
+                }
+                
+                if let data = data, let urlString = String(data: data, encoding: .utf8) {
+                    completion(urlString)
```

Comment:
the completion above is dispatched to the main queue but this one isn't. Could this cause a problem?